### PR TITLE
feat: add cryptoJS message id and itemChat hook typing

### LINKED ISSUE #136: Change crypto to crypto-js
Change dependency to node module crypto and use crypto-js instead.

crypto is required to compute sha1 of item keys

### PR DESCRIPTION
**Pull request by spaenleh**
closes #136

### CODE REVIEW

File: src/hooks/chat.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export default (
       return useQuery({
         queryKey: buildItemChatKey(itemId),
         queryFn: () =>
-          Api.getItemChat(itemId, queryConfig).then((data) => Map(data)),
+          Api.getItemChat(itemId, queryConfig).then((data) => Map<unknown, string|ChatMessage[]>(data)),
```
Comment (pyphilia) [inline src/hooks/chat.ts:35]: Do we still need that?

File: src/utils/item.ts [hunk 2]
```diff
@@ -46,14 +46,7 @@ export const getDirectParentId = (path: string) => {
...
 
-export const hashItemsIds = (ids?: UUID[]) =>
-  ids
-    ? crypto
-        .createHash('sha1')
-        .update([...ids].sort().join(''))
-        .digest('hex')
-        .toString()
-    : undefined;
+export const hashItemsIds = (ids?: UUID[]) => ids ? CryptoJS.SHA1([...ids].sort().join("")).toString() : undefined;
```
Comment (pyphilia) [inline src/utils/item.ts:49]: I think you can just go with `join()` (without empty string)

**Review by pyphilia — APPROVED** [review]
🌷

File: src/hooks/chat.ts (hunk 1 above)
Comment (spaenleh) [reply src/hooks/chat.ts]: I removed it in 68b7524 and the Chatbox did not complain. I think we can remove it.

**Pull request merged**